### ISSUE TITLE
Wrong @specs for Timezone.convert

### ISSUE THREAD

**Issue by sztosz** [bitwalker/timex#495]
**TL;DR**
In short, you can pass integer to `Timex.now()` and it will will gladly accpet integers from range -99 to 99, but if you do so, dialyzer will scream at you, because of incorrect `Timex.convert()` specs.

**Explanation**
`Timex.now(tz)` https://github.com/bitwalker/timex/blob/master/lib/timex.ex#L57 accepts `Types.valid_datetime` which are `String.t | tz_offset | :utc | :local` with `tz_offset` being `@type tz_offset :: -14..12`

`tz` is then passed to `Timex.convert()` https://github.com/bitwalker/timex/blob/master/lib/timezone/timezone.ex#L396 with it's specs
```
@spec convert(date :: DateTime.t, tz :: AmbiguousTimezoneInfo.t) :: AmbiguousDateTime.t | {:error, term}
@spec convert(date :: DateTime.t, tz :: TimezoneInfo.t | String.t) :: DateTime.t | AmbiguousDateTime.t | {:error, term}
```

The problem is that according to spec `tz` in spec of `convert` can't be `tz_offset` while later down the line:

`convert()` can call `do_get(tz, date, :utc)` https://github.com/bitwalker/timex/blob/master/lib/timezone/timezone.ex#L397 and while `do_get()` is private function, it indeed can call `name_of(timezone)` https://github.com/bitwalker/timex/blob/master/lib/timezone/timezone.ex#L214 where timezone is our `tz` from initial `Timex.now(tz)`. And `name_of` https://github.com/bitwalker/timex/blob/master/lib/timezone/timezone.ex#L70 does accept again this `timezone` or our initial `tz` being `tz_offset`. 

I though about tackling this problem in a PR to Timex, but I'm not sure, what will be correct way. I belive there is more to it that changing specs, see example below, I don't think -99 etc should be accepted at timezone, but I may be in the wrong.

--------------------
Example of `Timex.now()` working with integers.
```
iex(11)> Timex.now(0)  
#DateTime<2018-12-27 16:20:15.995290Z>
iex(12)> Timex.now(-99)
#DateTime<2018-12-23 13:20:19.933500-99:00 -99 Etc/GMT+99>
iex(13)> Timex.now(+99)
#DateTime<2018-12-31 19:20:24.130021+99:00 +99 Etc/GMT-99>
```

**Comment by bitwalker** [comment]
You are correct that we shouldn't allow arbitrary integers there, as obviously the range is constrained to a 24-hour day starting at -12. A PR is very welcome, definitely let us know if you have additional questions!
